src/Geometry/Primitives/NonCSG/ConvexPolygon.jl

```diff
@@ -0,0 +1,104 @@
+# MIT license
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See LICENSE in the project root for full license information.
+
+import Statistics
+import GeometricalPredicates
+
+"""
+    ConvexPolygon{T} <: Surface{T}
+
+General Convex Polygon surface, not a valid CSG object.
+The rotation of the polygon around its normal is defined by `rotationvec`.
+`rotationvec×surfacenormal` is taken as the vector along the u axis.
+
+```julia
+ConvexPolygon(local_frame::Transform{T}, local_polygon_points::Vector{Vector{T}}, interface::NullOrFresnel{T} = nullinterface(T))
+```
+
+The local frame defines the plane (spans by the right and up vectors) with the plane normal given by the forward vector.
+the local_polygon_points are given with respect to the local frame and are 2D points.
+"""
+struct ConvexPolygon{T} <: Surface{T}
+    plane::Plane{T,3}
+    local_frame::Transform{T}
+    local_points::Vector{Vector{T}}
```

Comment:
local_points should probably be SVector{SVector}. Vectors will be allocated on the heap which will lead to allocations and will probably also be slower. If we expected large convex polygons then Vector might be appropriate. But this class is going to be used for polygons with less than 20-30 points. If you change to SVector probably should document that should only have small number of points.

Reply:
I replaced it with Vector{SVector{2, T}}.
I can't find a convenient way to define an static array of static arrays. Please send me an example if you have one.

Reply:
you'd have to do something like this:

struct ConvexPolygon{N,T<:Real} <: Surface{T}
        local_points::SVector{N,SVector{2,T}}

or maybe even better

struct ConvexPolygon{Npoints,Dim,T<:Real} <: Surface{T}
        local_points::SVector{Npoints,SVector{Dim,T}}

where Dim is the dimension of the points, and Npoints is the number of points.

Reply:
@rgal does my suggestion for defined SVector{SVector} work for this case?

Reply:
I'm getting the following error when trying it:
ERROR: LoadError: TypeError: in ConvexPolygon, in N, expected N<:Int64, got a value of type Int64.
this is my entire test:
using StaticArrays

struct ConvexPolygon{N<:Int64, T<:Real} 
    local_points::SVector{N, SVector{2, T}}

    function ConvexPolygon(points::SVector{N, SVector{2, T}}) where {N<:Int64, T<:Real}
        new{N, T}(points)
    end

end

points = SVector(SVector(1.0, 1.0), SVector(2.0, 2.0), SVector(3.0, 3.0))
p = ConvexPolygon{3, Float64}(points)

Reply:
You don't want this:
```
struct ConvexPolygon{N<:Int64,T<:Real}
```
you want this:
```
struct ConvexPolygon{N,T<:Real}
```
The first statement is saying you want N to be a type which inherits from Int64, which is the same as saying a type equal to Int64 because there are no subtypes of Int64. But the argument you will use in the constructor is a number, not a type. 

you'll want to get rid of the N<:Int64 in the ConvexPolygonn constructor as well. I made those two changes and the code worked. This should do it:

```
struct ConvexPolygon{N, T<:Real} 
    local_points::SVector{N, SVector{2, T}}

    function ConvexPolygon(points::SVector{N, SVector{2, T}}) where {N, T<:Real}
        new{N, T}(points)
    end

end
```